src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -947,87 +992,141 @@ static RegexNode CloneMultiWithoutFirstChar(RegexNode node)
                 void EmitAllBranches()
                 {
                     // Label to jump to when any branch completes successfully.
-                    string doneAlternateLabel = ReserveName("Match");
+                    string matchLabel = ReserveName("AlternationMatch");
```

Comment:
Small Nit, but when you have an alternation which can backtrack, it results in code that looks like:

![image](https://user-images.githubusercontent.com/13854455/144511815-c876e91f-cee4-4428-a467-4e056d11b73f.png)

Maybe it's just me but AlternationMatch and AlternationBranch seem two very similar labels and for some reason my brain is mapping them as the same. should we consider using Snake_Case instead for the labels since we are already using underscores for local names?

Reply:
Sure, we can change it.  I expect there's a whole swath of stuff we'll want to look at for readability, in particular around naming.  I'll leave it as is for now but we can certainly rename these all subsequently.